packages/widget_toolkit/lib/src/lib_ui_components/buttons/small_button.dart

```diff
@@ -122,6 +122,31 @@ class SmallButton extends StatelessWidget {
           onPressed: onPressed,
           child: icon,
         );
+      case SmallButtonType.icon:
+        return Material(
+          shape: CircleBorder(
+            side: BorderSide(
+              color: context.widgetToolkitTheme.smallButtonBackgroundColor,
+              width: 0,
+            ),
+          ),
+          color: primaryColor,
+          child: Ink(
+            decoration: ShapeDecoration(
+              color: context.widgetToolkitTheme.smallButtonBackgroundColor,
+              shape: const CircleBorder(),
+            ),
+            child: IconButton(
```

Comment:
What's the purpose of the widgets above ? Do we really need them ?

Reply:
Yes, those are used in order to change the background color of the `IconButton`
`IconButton` has a property `color` but it's used to change the color of the icon.

Example of colored IconButton here
https://api.flutter.dev/flutter/material/IconButton-class.html#material.IconButton.2

Reply:
Please add golden tests with a two colors in light and dark mode

Reply:
This does nothing to the button unfortunately
Changing both background/foreground color doesn't change the background of the button

Reply:
Looking at the official documentation it looks like we have added extra widgets. 

Let's try to control the following colors

1. Background color
2. Icon color

**Official documentation**

```dart
 Material(
      color: Colors.white,
      child: Ink(
          decoration: const ShapeDecoration(
            color: Colors.red, //background color
            shape: CircleBorder(),
          ),
          child: IconButton(
            icon: const Icon(Icons.android),
            color: Colors.white, // icon color
          ),
        ),
    );
```

Reply:
It seems like it isn't necessary to use Material and Ink widgets after all, the default styling was being overwritten by by child widget of the IconButton